File: src/main/java/org/dxworks/githubminer/dto/export/UserDTO.java
package org.dxworks.githubminer.dto.export;

import com.google.api.client.util.Key;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.dxworks.githubminer.dto.commons.User;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {
    @Key
    private Long id;
    @Key
    private String login;
    @Key
    private String url;
    @Key
    private String email;
    @Key
    private String name;
    @Key
    private String avatarUrl;

    public static UserDTO fromUser(User user) {
        if (user == null)
            return null;
        return UserDTO.builder()
                .avatarUrl(user.getAvatarUrl())
                .email(user.getEmail())
                .login(user.getLogin())
                .id(user.getId())
                .url(user.getUrl())
                .name(user.getName())
                .avatarUrl(user.getAvatarUrl())
                .build();
    }
}

File: github-miner-app/src/main/kotlin/org/dxworks/githubminer/dto/export/UserDTO.kt
package org.dxworks.githubminer.dto.export

import com.google.api.client.util.Data
import com.google.api.client.util.Key
import org.dxworks.githubminer.dto.commons.User

class UserDTO {
    @Key
    var id: Long? = null

    @Key
    var login: String? = null

    @Key
    var url: String? = null

    @Key
    var email: String? = null

    @Key
    var name: String? = null

    @Key
    var avatarUrl: String? = null

    companion object {
        fun fromUser(user: User?): UserDTO? {
            return if (user == null || Data.isNull(user)) null else UserDTO().apply {
                    avatarUrl = user.avatarUrl
                    email = user.email
                    login = user.login
                    id = user.id
                    url = user.url
                    name = user.name
                    avatarUrl = user.avatarUrl
            }
        }
    }
}
